web/src/markdown.ts

```diff
@@ -85,26 +136,36 @@ function contains_problematic_linkifier({content, get_linkifier_map}) {
     return false;
 }
 
-function contains_topic_wildcard_mention(content) {
+function contains_topic_wildcard_mention(content: string): boolean {
     // If the content has topic wildcard mention (@**topic**) then don't
     // render it locally. We have only server-side restriction check for
     // @topic mention. This helps to show the error message (no permission)
     // via the compose banner and not to local-echo then fail due to restriction.
     return content.includes("@**topic**");
 }
 
-function content_contains_backend_only_syntax({content, get_linkifier_map}) {
+function content_contains_backend_only_syntax(
+    content: string,
+    get_linkifier_map: GetLinkifierMap,
+): boolean {
     // Try to guess whether or not a message contains syntax that only the
     // backend Markdown processor can correctly handle.
     // If it doesn't, we can immediately render it client-side for local echo.
     return (
         contains_preview_link(content) ||
-        contains_problematic_linkifier({content, get_linkifier_map}) ||
+        contains_problematic_linkifier(content, get_linkifier_map) ||
         contains_topic_wildcard_mention(content)
     );
 }
 
-function parse_with_options({raw_content, helper_config, options}) {
+function parse_with_options(
```

Comment:
Would changing the parameters to be not an object be better as a prep commit? 

(Would these three be okay as a combined commit? The changes are quite small)

Reply:
Possibly but it's definitely easier to read as individual prep commits and commits are cheap, so I'm not sure why one would prefer to do that.